caddy/caddy.go

```diff
@@ -391,6 +419,20 @@ func (FrankenPHPModule) CaddyModule() caddy.ModuleInfo {
 // Provision sets up the module.
 func (f *FrankenPHPModule) Provision(ctx caddy.Context) error {
 	f.logger = ctx.Slogger()
+	app, err := ctx.App("frankenphp")
+	if err != nil {
+		return err
+	}
+	fapp, ok := app.(*FrankenPHPApp)
+	if !ok {
+		return fmt.Errorf(`expected ctx.App("frankenphp") to return FrankenPHPApp, got %T`, app)
+	}
```

Comment:
I think this should probably be something like this:

```suggestion
	app, err := ctx.App("frankenphp")
	if err != nil {
		return err
	}
	fapp, ok := app.(*FrankenPHPApp)
	if !ok || fapp == nil {
		return fmt.Errorf(`expected ctx.App("frankenphp") to return *FrankenPHPApp, got %T`, app)
	}
```

IIRC, if `app` is `nil`, trying to assert the type will still retain `nil`. So, we need to check for `nil` as well. Also, it is expecting a pointer, not a value. So the error message should reflect that.